### ISSUE TITLE
Breaking changes in 1.1.0-beta3 with AddLegacySource method

### ISSUE THREAD

**Issue by utpilla** [open-telemetry/opentelemetry-dotnet#2051]
With the 1.1.0-beta3 there are breaking changes with the use of `AddLegacySource` in the Instrumentation packages.

`AddLegacySource` method was moved from `TracerProviderBuilderExtensions.cs` in SDK to API in 1.1.0-beta3.

1.0.0-rc4 version of the instrumentation packages expect `AddLegacySource` method to be present in the SDK and they fail when used with an SDK version 1.1.0-beta3. Up until 1.1.0-beta2, `AddLegacySource` method was in the SDK.

We need to release new packages for API + SDK (1.1.0-beta4) and for instrumentation packages (1.0.0-rc5).

**Comment by Mario-Hofstaetter** [comment]
This issue is about `OpenTelemetry.Exporter.Jaeger` correct? 

Just some context for folks google'ing the issue,
with `1.1.0-beta3` (other packages on `1.0.0-rc4`) im getting this on startup with ASP.NET Core 3.1 (linked #2050 mentions asp.net core 5.0 ?)

```text
Unhandled exception. System.MissingMethodException: Method not found: 'OpenTelemetry.Trace.TracerProviderBuilder OpenTelemetry.Trace.TracerProviderBuilderExtensions.AddLegacySource(OpenTelemetry.Trace.TracerProviderBuilder, System.String)'.
   at OpenTelemetry.Trace.TracerProviderBuilderExtensions.AddHttpClientInstrumentation(TracerProviderBuilder builder, Action`1 configureHttpClientInstrumentationOptions)
   at otelaspnetcoretest.Startup.<>c__DisplayClass4_0.<ConfigureServices>b__0(TracerProviderBuilder builder) in D:\xxxxxxxxx\otelaspnetcoretest\otelaspnetcoretest\Startup.cs:line 26
   at Microsoft.Extensions.DependencyInjection.OpenTelemetryServicesExtensions.AddOpenTelemetryTracing(IServiceCollection services, Action`1 configure)
   at otelaspnetcoretest.Startup.ConfigureServices(IServiceCollection services) in D:\xxxxxxxxx\otelaspnetcoretest\otelaspnetcoretest\Startup.cs:line 26
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.InvokeCore(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass9_0.<Invoke>g__Startup|0(IServiceCollection serviceCollection)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.Invoke(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass8_0.<Build>b__0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.UseStartup(Type startupType, HostBuilderContext context, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.<>c__DisplayClass12_0.<UseStartup>b__0(HostBuilderContext context, IServiceCollection services)
   at Microsoft.Extensions.Hosting.HostBuilder.CreateServiceProvider()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at otelaspnetcoretest.Program.Main(String[] args) in D:\xxxxxxxxx\otelaspnetcoretest\otelaspnetcoretest\Program.cs:line 16

```

**Comment by utpilla** [comment]
@Mario-Hofstaetter This issue is not about `OpenTelemetry.Exporter.Jaeger`. The issue is with Instrumentation libraries such as `OpenTelemetry.Instrumentation.Http` and `OpenTelemetry.Instrumentation.AspNetCore` that are calling the `AddLegacySource` method.

**Comment by ronaldvocus** [comment]
Has anyone addressed this yet or its still pending. Trying to do some testing on our systems and ran into this issue. Are there any workarounds until its fixed?

**Comment by purkhusid** [comment]
Having issues with this when I tried to use the `OpenTelemetry.Contrib.Preview` package which depends on `OpenTelemetry` `1.0.0-beta3` or greater which does not work with `OpenTelemetry.Instrumentation.Http`

**Comment by ronaldvocus** [comment]
@purkhusid Use the stable version of open telemetry which is version 1.0. 1.1 is still in development and will cause issues as things break.

**Comment by purkhusid** [comment]
I would use the 1.0 but I wanted to try the `OpenTelemetry.Contrib.Preview` package which is incompatible with 1.0.

**Comment by utpilla** [comment]
We have released the newer versions of core components (1.1.0-beta4) and instrumentations (1.0.0-rc5). Please update the package versions to fix this issue.
